yb-voyager/cmd/live_migration.go

```diff
@@ -106,21 +110,87 @@ func (event *Event) getDeleteStmt(targetSchema string) string {
 //==================================================================================
 
 func streamChanges(connPool *tgtdb.ConnectionPool, targetSchema string) error {
-	queueFilePath := filepath.Join(exportDir, "data", "queue.json")
-	log.Infof("Streaming changes from %s", queueFilePath)
-	file, err := os.OpenFile(queueFilePath, os.O_CREATE, 0640)
+	cdcDirPath := filepath.Join(exportDir, "data", "cdc")
+	archiveDirPath := filepath.Join(exportDir, "data", "cdc", "archive")
+	err := os.MkdirAll(archiveDirPath, 0755)
 	if err != nil {
-		return fmt.Errorf("error opening file %s: %v", queueFilePath, err)
+		return fmt.Errorf("error creating archive dir: %v", err)
+	}
+
+	// keep looking for segments continuously
+	// and stream changes from segments one by one in sequence
+	for {
+		dirEntries, err := os.ReadDir(cdcDirPath)
+		if err != nil {
+			return fmt.Errorf("error reading cdc dir: %v", err)
+		}
+
+		var queueSegmentFiles []string
+		for _, dirEntry := range dirEntries {
+			if !dirEntry.IsDir() && strings.HasSuffix(dirEntry.Name(), ".ndjson") {
+				queueSegmentFiles = append(queueSegmentFiles, dirEntry.Name())
+			}
+		}
+
+		// sort files by segment number
+		sort.Slice(queueSegmentFiles, func(i, j int) bool {
+			segNumIdx := len("queue.")
+			len1 := len(queueSegmentFiles[i])
+			len2 := len(queueSegmentFiles[j])
+
+			segNum1, err := strconv.ParseInt(queueSegmentFiles[i][segNumIdx:len1-7], 10, 64)
+			if err != nil {
+				utils.ErrExit("error parsing segment file name for segNum1: %v", err)
+			}
+
+			segNum2, err := strconv.ParseInt(queueSegmentFiles[j][segNumIdx:len2-7], 10, 64)
+			if err != nil {
+				utils.ErrExit("error parsing segment file name for segNum2: %v", err)
+			}
+
+			return segNum1 < segNum2
+		})
+
+		// walk through the segment files in cdc dir
+		for _, queueSegmentFile := range queueSegmentFiles {
+			queueFilePath := filepath.Join(cdcDirPath, queueSegmentFile)
+			log.Infof("streaming changes from %s", queueFilePath)
+			err = streamChangesFromFile(connPool, queueFilePath, targetSchema)
+			if err != nil {
+				return fmt.Errorf("error streaming changes from %s: %v", queueSegmentFile, err)
+			}
+
+			// move queue file to archive directory once streamed
+			archivePath := filepath.Join(archiveDirPath, filepath.Base(queueSegmentFile))
+			err = os.Rename(queueFilePath, archivePath)
+			if err != nil {
+				return fmt.Errorf("error archiving file %s: %v", queueFilePath, err)
+			}
+		}
+		time.Sleep(time.Second * 2)
+	}
+}
+
+func streamChangesFromFile(connPool *tgtdb.ConnectionPool, path string, targetSchema string) error {
+	file, err := os.OpenFile(path, os.O_RDONLY, 0640)
+	if err != nil {
+		return fmt.Errorf("error opening file %s: %v", path, err)
 	}
 	defer file.Close()
 
+	eventsCount := int64(0)
 	r := utils.NewTailReader(file)
-	dec := json.NewDecoder(r)
-	log.Infof("Waiting for changes in %s", queueFilePath)
-	// TODO: Batch the changes.
-	for dec.More() {
```

Comment:
I have requested @makalaaneesh to fallback to the older way of writing events. The events should be written in pretty format. We should NOT use the "newline delimited json".